### PR TITLE
Add %empty-directory substitution

### PR DESCRIPTION
**Pull request by CodaFi**
This rebases #4616 but does not update any of the tests that have since changed.

Let's see what happens.

### CODE REVIEW

**Comment by CodaFi** [comment]
@swift-ci please clean test

File: test/ClangImporter/MixedSource/broken-modules.swift [hunk 1]
```diff
@@ -1,5 +1,10 @@
+<<<<<<< HEAD:test/ClangImporter/MixedSource/broken-modules.swift
```
Comment (CodaFi) [inline test/ClangImporter/MixedSource/broken-modules.swift]: Well, this is embarrassing...

File: validation-test/compiler_crashers/28482-hasaccessibility-accessibility-not-computed-yet.swift [hunk 2]
```diff
@@ -7,4 +7,14 @@
 
 // RUN: not --crash %target-swift-frontend %s -emit-ir
 // REQUIRES: asserts
-{guard let{protocol A{extension{let d}}}let f=d
```
Comment (gparker42) [inline validation-test/compiler_crashers/28482-hasaccessibility-accessibility-not-computed-yet.swift]: Is this change intended?

File: docs/Testing.rst [hunk 3]
```diff
@@ -397,6 +397,10 @@ Other substitutions:
 
 * ``%raw-FileCheck``: the LLVM ``FileCheck`` utility.
 
+* ``%empty-directory(`` *directory-name* ``)``: ensures that the given
+  directory exists and is empty.  Equivalent to
+  ``rm -rf directory-name && rm -rf directory-name``.
```
Comment (gparker42) [inline docs/Testing.rst]: typo: rm -rf && mkdir -p

File: test/lit.cfg [hunk 4]
```diff
@@ -1014,6 +1014,9 @@ config.substitutions.append(('%rth', config.rth))
 config.substitutions.append(('%scale-test',
                              '{} --swiftc-binary={} --tmpdir=%t'.format(
                                  config.scale_test, config.swiftc)))
+config.substitutions.append(
+    ('%empty-directory\(([^)]+)\)', 'rm -rf \\1 && mkdir -p \\1'))
```
Comment (gparker42) [inline test/lit.cfg]: Does this need quoting to defend against paths with spaces in them?

File: validation-test/compiler_crashers/28482-hasaccessibility-accessibility-not-computed-yet.swift (hunk 2 above)
Comment (CodaFi) [inline validation-test/compiler_crashers/28482-hasaccessibility-accessibility-not-computed-yet.swift]: Absolutely not.  There is a fix for a crasher coming our way tho 😀

**Comment by CodaFi** [comment]
@swift-ci test

**Comment by CodaFi** [comment]
@swift-ci test

File: test/lit.cfg (hunk 4 above)
Comment (CodaFi) [inline test/lit.cfg]: I believe that was the original intent.  Is it overkill?

**Comment by CodaFi** [comment]
@swift-ci please test

**Comment by CodaFi** [comment]
@swift-ci please test

**Comment by CodaFi** [comment]
@swift-ci please test

**Comment by CodaFi** [comment]
@swift-ci please test

**Comment by CodaFi** [comment]
⛵️
